plugins/lockToVote/hooks/useProposalVeto.tsx

```diff
@@ -73,12 +104,58 @@ export function useProposalVeto(proposalId: string) {
   }, [vetoingStatus, vetoTxHash, isConfirming, isConfirmed]);
 
   const vetoProposal = () => {
-    vetoWrite({
-      abi: LockToVetoPluginAbi,
-      address: PUB_LOCK_TO_VOTE_PLUGIN_ADDRESS,
-      functionName: "veto",
-      args: [proposalId, 50000000000000000000],
-    });
+    if (!proposal || !balanceResult || !nonceResult || !nameResult || !versionResult) return;
+    const balance = BigInt(Number(balanceResult?.result));
+    const nonce = BigInt(Number(nonceResult?.result));
+    const erc20_name = nameResult?.result;
+    const versionFromContract = versionResult?.result;
+
+    let deadline = BigInt(Math.floor(Date.now() / 1000) + 60 * 60);
+    let value = balance; 
+    let destination: `0x${string}` = PUB_LOCK_TO_VOTE_PLUGIN_ADDRESS;
+
+    const domain = {
+      chainId: PUB_CHAIN.id,
+      name: String(erc20_name),
+      /* We assume 1 if permit version is not specified */
+      version: String(versionFromContract ?? '1'),
+      verifyingContract: PUB_TOKEN_ADDRESS,
+    };
+
+    const types = {
```

Comment:
Move this type to the types file in the /utils at the root of Aragonette

Reply:
I'm not sure if it makes sense in this case. It's not a TS type. It's just the Solidity type names to encode the permit message.